lib/agreements.js

```diff
@@ -1,42 +1,97 @@
 'use strict';
 
 var _ = require('lodash');
+var Joi = require('joi');
 
 exports.register = function (server, options, next) {
-  options = _.extend({ basePath: '/api/2.0' }, options);
+  options = _.extend({basePath: '/api/2.0'}, options);
   var handlers = require('./handlers.js')(server, 'cd-agreements');
 
   server.route([{
     method: 'POST',
     path: options.basePath + '/agreements',
     handler: handlers.actHandler('save'),
     config: {
-      description: 'Save the agreement',
-      tags: ['api', 'users']
+      description: 'Saves the specified agreement',
+      tags: ['api', 'users'],
+      plugins: {
+        'hapi-swagger': {
+          responseMessages: [
+            {code: 400, message: 'Bad Request'},
+            {code: 200, message: 'OK'}
+          ]
+        }
+      },
+      validate: {
+        payload: Joi.object({
+          agreement: {
+            userId: Joi.alternatives().try(Joi.string().guid(), Joi.string()).required()
+          }
+        })
+      }
     }
   }, {
     method: 'POST',
     path: options.basePath + '/agreements/list-by-ids',
-    handler: handlers.actHandler('get_agreements'),
+    handler: handlers.actHandler('list'),
     config: {
-      description: 'List Agreement by Id',
-      tags: ['api', 'users']
+      description: 'List Agreements by Id',
+      tags: ['api', 'users'],
+      plugins: {
+        'hapi-swagger': {
+          responseMessages: [
+            {code: 400, message: 'Bad Request'},
+            {code: 200, message: 'OK'}
+          ]
+        }
+      },
+      validate: {
+        payload: Joi.object({
+          userIds: Joi.array(Joi.string())
+        })
+      }
     }
   }, {
     method: 'POST',
     path: options.basePath + '/agreements/count',
     handler: handlers.actHandler('count'),
     config: {
       description: 'Count the number of agreements',
-      tags: ['api', 'users']
+      tags: ['api', 'users'],
+      plugins: {
+        'hapi-swagger': {
+          responseMessages: [
+            {code: 400, message: 'Bad Request'},
+            {code: 200, message: 'OK'}
+          ]
+        }
+      },
+      validate: {
+        payload: Joi.object({ query: {
```

Comment:
This is used by the stats page in /dashboard/stats (login as manager1).
The current payload used is {query : {agreement_version: 2}} because we mostly care about the latest version of the agreement :)